### DISCUSSION TITLE
Pruning Models command

### DISCUSSION THREAD

**Discussion by mahmoudmohamedramadan** [laravel/framework#38900] [Q&A] [answered]
while I was training on the `Pruning Models` feature and BY luck I wrote `php artisan model:prun --model=\App\Models\Dog` instead of `php artisan model:prune --model=\App\Models\Dog` and the strange that this command has worked 😂 despite the command that Laravel support is `model:prune` I want to know how this happened?

![{F022F992-C517-47BF-97AC-126D43F5F6F2}](https://user-images.githubusercontent.com/48416569/134292028-02946976-0c55-4d53-8d76-9984dbfc9f11.png)

**Comment by dennisprudlo** [comment, marked as answer]
The symfony commands detect commands in the same ways as the autocomplete in a terminal.

```shell
php artisan route:clear
php artisan route:clea
php artisan route:cle
php artisan route:cl
```

All of these will execute `route:clear` because there is only one possibility for `route:cl`. But when the name is ambiguous symfony will show an error:
```shell
php artisan route:c

  Command "route:c" is ambiguous.                                          
  Did you mean one of these?                                               
      route:cache Create a route cache file for faster route registration  
      route:clear Remove the route cache file. 
```

And since `prune` is the only command in the `model` namespace you can actually run `php artisan model:` and it will prune the models

**Reply by mahmoudmohamedramadan** [reply]
Great! thanks
